hledger-lib/Hledger/Data/Types.hs

```diff
@@ -214,6 +214,7 @@ instance Eq Posting where
 -- | The position of parse errors (eg), like parsec's SourcePos but generic.
 -- File name, 1-based line number and 1-based column number.
 data GenericSourcePos = GenericSourcePos FilePath Int Int
+                      | LedgerSourcePos FilePath (Int, Int) -- ^ range of lines in ledger journal
```

Comment:
> Am I understanding this commit rightly: it gives us the option to report a range of lines, rather than a single line and column position, in parse errors ?

You right. But referring the range is only the part of the story. Note that I intentionally ignore columns since they'll refer full lines for ledger/journal entries.
It makes a special case of `GenericSourcePos` that valid only for "Ledger" journal (both `hledger` and C++ alternative `ledger`). The reason to distinguish is that we know only how to generate such kind of transaction entries. I.e. we cannot make a good in-line patch for "time log" journals simply because only some subset of all possible `Transaction` entries may be represented in a "time log" entry. Adding an extra posting probably will kick out `Transaction` read from "time log" from that subset.

> Your new usage seems a reason to keep it around though, perhaps with a better name ?

Yes. I don't really like this name. Especially after adding a new ctor.

> The haddock needs an update.

True. I'll update this.

> I'd prefer to use "Journal" instead of "Ledger"; that's the term used elsewhere.

Agree. Will update.